pkg/collector/corechecks/oracle-dbm/statements.go

```diff
@@ -398,51 +452,96 @@ func (c *Check) StatementMetrics() (int, error) {
+					queryRow.Tables = obfuscatedStatement.tables
+					SQLStatement = obfuscatedStatement.statement
+					found = true
+				}
 			}
+			if !found {
+				if statementMetricRow.StatementMetricsKeyDB.ForceMatchingSignature == "" {
+					queryHashCol = "sql_id"
+				} else {
+					queryHashCol = "force_matching_signature"
+				}
 
-			var SQLStatement string
-			rows.Next()
-			cols, err := rows.SliceScan()
-			rows.Close()
-			if err != nil {
-				log.Errorf("query metrics statement scan error %s %s %+v", err, SQLTextQuery, p)
-				SQLTextErrors++
-				continue
-			}
-			SQLStatement = cols[0].(string)
-			sender.Histogram("dd.oracle.statements_metrics.sql_text_length", float64(len(SQLStatement)), "", c.tags)
+				p := map[string]interface{}{
+					"force_matching_signature": statementMetricRow.StatementMetricsKeyDB.ForceMatchingSignature,
+					"sql_id":                   statementMetricRow.StatementMetricsKeyDB.SQLID,
+				}
+				SQLTextQuery := fmt.Sprintf("SELECT /* DD */ sql_fulltext FROM v$sqlstats WHERE %s=:%s AND rownum = 1", queryHashCol, queryHashCol)
 
-			queryRow := QueryRow{}
-			//obfuscatedStatement, err := c.GetObfuscatedStatement(o, SQLStatement, statementMetricRow.ForceMatchingSignature, statementMetricRow.SQLID)
-			obfuscatedStatement, err := c.GetObfuscatedStatement(o, SQLStatement)
-			SQLStatement = obfuscatedStatement.Statement
-			if err == nil {
-				queryRow.QuerySignature = obfuscatedStatement.QuerySignature
-				queryRow.Commands = obfuscatedStatement.Commands
-				queryRow.Tables = obfuscatedStatement.Tables
-			}
+				rows, err := c.db.NamedQuery(SQLTextQuery, p)
+				if err != nil {
+					log.Errorf("query metrics statements error named exec %s %s %+v", err, SQLTextQuery, p)
+					SQLTextErrors++
+					continue
+				}
+				defer rows.Close()
+
+				rows.Next()
+				cols, err := rows.SliceScan()
+				totalSQLTextTimeUs += time.Since(startSQLText).Microseconds()
+
+				if err != nil {
+					log.Errorf("query metrics statement scan error %s %s %+v", err, SQLTextQuery, p)
+					SQLTextErrors++
+					continue
+				}
+				SQLStatement = cols[0].(string)
+
+				obfuscatedStatement, err := c.GetObfuscatedStatement(o, SQLStatement)
+				SQLStatement = obfuscatedStatement.Statement
+				if err == nil {
+					queryRow.QuerySignature = obfuscatedStatement.QuerySignature
+					queryRow.Commands = obfuscatedStatement.Commands
+					queryRow.Tables = obfuscatedStatement.Tables
 
+					if c.config.InstanceConfig.QueryMetrics.IncludeDatadogQueries {
+						cacheEntry := StatementsCacheData{
+							statement:      SQLStatement,
+							querySignature: obfuscatedStatement.QuerySignature,
+							tables:         obfuscatedStatement.Tables,
+							commands:       obfuscatedStatement.Commands,
+						}
+						if statementMetricRow.StatementMetricsKeyDB.ForceMatchingSignature != "" {
+							c.DDstatementsCache.forceMatchingSignatures[statementMetricRow.StatementMetricsKeyDB.ForceMatchingSignature] = cacheEntry
```

Comment:
is this the correct cache to store this new entry in? I would think it should be this one `c.statementsCache.forceMatchingSignatures`?